mllib/src/main/scala/org/apache/spark/mllib/linalg/Matrices.scala

```diff
@@ -278,7 +278,8 @@ class DenseMatrix @Since("1.3.0") (
   }
 
   override def hashCode: Int = {
-    com.google.common.base.Objects.hashCode(numRows : Integer, numCols: Integer, toArray)
+    val state = Seq(numRows, numCols, Arrays.hashCode(values), isTransposed.hashCode)
```

Comment:
Good to double-check that. I think we have to write a `hashCode` that is consistent with Breeze's `equals` for now and leave a to-do to change it to Breeze's later. Although it sounds brittle, it seems like equality must be defined in the obvious way -- equal elements everywhere. The hash code can just be a function of the elements since equality must be. It may need to be a function of non-zero elements and their positions, in order, in order to be efficient for sparse matrices.